### ISSUE TITLE
v1.16.0 SyntaxError: Identifier 'process' has already been declared

### ISSUE THREAD

**Issue by bryce-pearce** [remix-run/remix#6280]
### What version of Remix are you using?

1.16.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

Not sure the root cause, but this issue is appearing after upgrading Remix from 1.15 to 1.16, no other changes were made.

<img width="498" alt="image" src="https://user-images.githubusercontent.com/78380005/235750946-2b6e29d1-b4cd-40d4-b5e9-f323591d7ba9.png">

<img width="423" alt="image" src="https://user-images.githubusercontent.com/78380005/235744397-989abbf1-5574-45f4-afc0-b736816431b2.png">

**Note**: Going through versions and it appears this issue has been occurring since at least `1.16.0-pre.4`. But from that version until earlier versions have this additional error:
`Cannot destructure property 'basename' of 'React__namespace.useContext(...)' as it is null.`

### Expected Behavior

There is no crash when upgrading to 1.16

### Actual Behavior

There is a crash when upgrading to 1.16

**Comment by pcattori** [comment]
@bryce-pearce could you share what (if any) future flags you have set in your `remix.config.js`?

**Comment by bryce-pearce** [comment]
> @bryce-pearce could you share what (if any) future flags you have set in your `remix.config.js`?

None, my `remix.config.js` looks like
```js
module.exports = {
  serverDependenciesToBundle: [/^lodash-es.*/],
  ignoredRouteFiles: ['.*'],
  devServerBroadcastDelay: 1000,
};
```

**Comment by pcattori** [comment]
Probably related to https://github.com/remix-run/remix/pull/5274 and https://github.com/remix-run/remix/pull/6259

**Comment by bryce-pearce** [comment]
if anyone has updates, suggestions or workarounds, it would be much appreciated!

**Comment by HossamMarey** [comment]
If anyone has updates, suggestions, or workarounds regarding the same issue, I would greatly appreciate it!

**Comment by MichaelDeBoey** [comment]
We just published version `1.17.1-pre.0` which includes #6562 (a possible fix). If you'd like to take it for a test run please try it out and let us know what you think!
